File: python/VMPrototype1/compilation_scope.py
from typing import Optional

from emitted_instruction import EmittedInstruction
from instructions import Instructions


class CompilationScope:
    def __init__(self, instructions: Instructions = None,
                 last_instruction: EmittedInstruction = None,
                 previous_instruction: EmittedInstruction = None):
        self.instructions = instructions
        self.last_instruction: Optional[EmittedInstruction] = last_instruction
        self.previous_instruction: Optional[EmittedInstruction] = previous_instruction

File: cpp/vm/advm/compilation_scope.cpp
#include "compilation_scope.h"

CompilationScope::CompilationScope() {
    instructions = Instructions();
    previousInstruction = EmittedInstruction();
    lastInstruction = EmittedInstruction();
}

CompilationScope::CompilationScope(Instructions i) {
    instructions = i;
    previousInstruction = EmittedInstruction();
    lastInstruction = EmittedInstruction();
}

CompilationScope::CompilationScope(Instructions i, EmittedInstruction e1, EmittedInstruction e2) {
    instructions = i;
    previousInstruction = e1;
    lastInstruction = e2;
}